applications/nrf5340_audio/unicast_client/main.c

```diff
@@ -292,6 +292,40 @@ static void le_audio_msg_sub_thread(void)
 
 			break;
 
+		case LE_AUDIO_EVT_COORD_SET_DISCOVERED:
+			uint8_t num_conn = 0;
+			uint8_t num_filled = 0;
+
+			bt_mgmt_num_conn_get(&num_conn);
+
+			if (msg.set_size > 0) {
+				/* Check how many active connections we have for the given set */
+				LOG_DBG("Setting SIRK");
+				bt_mgmt_scan_sirk_set(msg.sirk);
+				bt_mgmt_set_size_filled_get(&num_filled);
+
+				LOG_INF("Set members found so far: %d/%d", num_filled,
+					msg.set_size);
+
+				if (num_filled == msg.set_size) {
+					/* All devices in set found, clear sirk before scanning */
+					bt_mgmt_scan_sirk_set(NULL);
```

Comment:
Why is this required?

Reply:
I may have missed it, but if one of the set members disconnect, where is this reset?

Reply:
When the sirk is set we will only accept devices from that set. When setting it to NULL we accept any device (like we did before). 

So the mechanism now is that when we connect to a device that is part of a set, we will only look for its partner. If it's not part of a set we will accept any device with the same device name